src/librbd/librbd.cc

```diff
@@ -1286,6 +1286,27 @@ namespace librbd {
     return r;
   }
 
+  ssize_t Image::writesame(uint64_t ofs, size_t len, bufferlist& bl, int op_flags)
+  {
+    ImageCtx *ictx = (ImageCtx *)ctx;
+    size_t data_len = bl.length();
+    tracepoint(librbd, writesame_enter, ictx, ictx->name.c_str(), ictx->snap_name.c_str(),
+               ictx->read_only, ofs, len, data_len > 0 ? NULL : bl.c_str(), data_len, op_flags);
+
+    if (len == 0) {
+      ictx->snap_lock.get_read();
+      len = ictx->get_image_size(ictx->snap_id) - ofs;
```

Comment:
Why would we want to default the write length to the (remaining) image size instead of letting the client tell us the length?